vector/src/test/java/im/vector/app/features/home/room/detail/timeline/factory/PollItemFactoryTest.kt

```diff
@@ -0,0 +1,282 @@
+                    ?.answers
+                    ?.mapToOptions(PollState.Sending, A_MESSAGE_INFORMATION_DATA)
+                    ?.forEachIndexed { index, pollOptionViewState ->
+                        A_POLL_CONTENT.getBestPollCreationInfo()?.answers?.get(index)?.let { option ->
+                            pollOptionViewState shouldBeEqualTo PollOptionViewState.PollSending(option.id ?: "", option.getBestAnswer() ?: "")
+                        }
+                    }
+        }
+    }
+
+    @Test
+    fun `given a sent poll when a vote is cast then all option view states is PollVoted`() = runTest {
+        with(pollItemFactory) {
+            A_POLL_CONTENT
+                    .getBestPollCreationInfo()
+                    ?.answers
+                    ?.mapToOptions(PollState.Voted(1), A_MESSAGE_INFORMATION_DATA)
+                    ?.forEachIndexed { index, pollOptionViewState ->
+                        A_POLL_CONTENT.getBestPollCreationInfo()?.answers?.get(index)?.let { option ->
+                            val voteSummary = A_MESSAGE_INFORMATION_DATA.pollResponseAggregatedSummary?.votes?.get(option.id)
+                            pollOptionViewState shouldBeEqualTo PollOptionViewState.PollVoted(
+                                    optionId = option.id ?: "",
+                                    optionAnswer = option.getBestAnswer() ?: "",
+                                    voteCount = A_MESSAGE_INFORMATION_DATA.pollResponseAggregatedSummary?.totalVotes ?: 0,
+                                    votePercentage = voteSummary?.percentage ?: 0.0,
+                                    isSelected = A_MESSAGE_INFORMATION_DATA.pollResponseAggregatedSummary?.myVote == option.id,
+                            )
+                        }
+                    }
+        }
+    }
+
+    @Test
+    fun `given a sent poll when the poll is undisclosed then all option view states is PollUndisclosed`() = runTest {
+        with(pollItemFactory) {
+            A_POLL_CONTENT
+                    .getBestPollCreationInfo()
+                    ?.answers
+                    ?.mapToOptions(PollState.Undisclosed, A_MESSAGE_INFORMATION_DATA)
+                    ?.forEachIndexed { index, pollOptionViewState ->
+                        A_POLL_CONTENT.getBestPollCreationInfo()?.answers?.get(index)?.let { option ->
+                            pollOptionViewState shouldBeEqualTo PollOptionViewState.PollUndisclosed(
+                                    optionId = option.id ?: "",
+                                    optionAnswer = option.getBestAnswer() ?: "",
+                                    isSelected = A_MESSAGE_INFORMATION_DATA.pollResponseAggregatedSummary?.myVote == option.id,
+                            )
+                        }
+                    }
+        }
+    }
+
+    @Test
+    fun `given an ended poll then all option view states is Ended`() = runTest {
+        with(pollItemFactory) {
```

Comment:
I would argue that the `optionViewStates` can exist on a parent `PollItemViewState`, if we want to avoid too many conditions under a single class, we can extract out another factory which the `PollItemViewStateFactory` could depend on


```kotlin
// Our PollItemViewState test doesn't need to know how the option view state is generated
// that class can have its own tests 
fakeOptionViewStateFactory.given(A_OPTION_VIEW_STATE)

val result = factory.create()

result shouldBeEqualTo Result(
  optionViewStates = A_OPTION_VIEW_STATE
)
```